olesia.olshevskaia/task-2-2/cmd/service/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"container/heap"
+	"fmt"
+)
+
+type intHeap []int
+
+func (h *intHeap) Len() int {
+	return len(*h)
+}
+
+func (h *intHeap) Less(i, j int) bool {
+	return (*h)[i] > (*h)[j]
+}
+
+func (h *intHeap) Swap(i, j int) {
+	(*h)[i], (*h)[j] = (*h)[j], (*h)[i]
+}
+
+func (h *intHeap) Push(x interface{}) {
+	newX, isInt := x.(int)
```

Comment:
<img width="400" height="763" alt="Image" src="https://github.com/user-attachments/assets/05975ad9-7658-48c8-bc49-6e58cf3710a7" />

```suggestion
	newX, ok:= x.(int)
```